Dependencies/xr/Source/ARCore/XR.cpp

```diff
@@ -495,6 +519,93 @@ namespace xr
     {
     }
 
+    void System::Session::Frame::GetHitTestResults(std::vector<Pose>& filteredResults, xr::Ray offsetRay) const
```

Comment:
I think we should put all this logic in Session::Impl itself since it has all the state, and have this function just delegate to the one in Session::Impl. This way we can keep all the state private in Session::Impl, and Frame is really just a thin proxy. That is the pattern I went with for DrawFrame, so I think it would make sense to replicate that here.

Reply:
That makes sense.  I'll move this into SessionImpl and move the member variables back to private.